### PR TITLE
Level completed door

### PR DESCRIPTION
**Pull request by tinaStremenova**
- Added door user can enter to complete level. 
- Door only unlocks when user killed all zombies
- Door opens when user is close to it and closes when user leaves it's proximity

https://user-images.githubusercontent.com/7025546/195110431-a08753a1-cbc5-492d-8f13-ecc35c3e4274.mov

### CODE REVIEW

File: SpriteKit/MendeluSpriteKit/Game/SceneObjects/Door.swift [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            timePerFrame: 0.2,
+            resize: false,
+            restore: false
+        )
+    }
+
+    func updateState() {
+        switch distanceToPlayer {
+        case let distance where distance <= 80:
+            guard !isOpen, isUnlocked else { return }
```
Comment (robha141) [inline SpriteKit/MendeluSpriteKit/Game/SceneObjects/Door.swift:112]: I would just add new line after guard, but it's optional 😀

File: SpriteKit/MendeluSpriteKit/Game/SceneObjects/Door.swift [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            playAnimation(key: Animations.open.rawValue)
+            isOpen = true
+
+            if physicsBody == nil {
+                physicsBody = collisionPhysicsBody
+            }
+
+        case let distance where distance > 80:
+            guard isOpen else { return }
```
Comment (robha141) [inline SpriteKit/MendeluSpriteKit/Game/SceneObjects/Door.swift:122]: also here, so that conditions are distinguished in code 😄

**Review by robha141 — APPROVED** [review]
🦇

File: SpriteKit/MendeluSpriteKit/Game/SceneObjects/Door.swift (hunk 1 above)
Comment (tinaStremenova) [reply SpriteKit/MendeluSpriteKit/Game/SceneObjects/Door.swift:112]: fixed

File: SpriteKit/MendeluSpriteKit/Game/SceneObjects/Door.swift (hunk 2 above)
Comment (tinaStremenova) [reply SpriteKit/MendeluSpriteKit/Game/SceneObjects/Door.swift:122]: fixed

**Pull request merged**